Feature/4 entity create

5nam opened the pull request:
Entity 생성 및 연관관계 맵핑 완료했습니다.

잘못된 맵핑, 테이블 등이 있다면 알려주시면 감사하겠습니다.

5nam committed:
✨ 생성 Entity

1. domain 은 주요 테이블을 기준으로 나눴습니다. Coupon/Member/Review/Store
2. 단방향 매핑만 했습니다.
3. Time 으로 묶어서 created_at, updated_at 을 관리합니다.

5nam committed:
✨ 추가 EnableJpaAuditing

5nam committed:
🩹 변경/추가 Entity 어노테이션
1. Coupon 테이블 이름 지정 삭제(불필요)
2. FavoriteStore 에 @Entity 어노테이션 추가

5nam committed:
🩹 추가 패키지 구조
1. 패키지 구조 반영을 위한 파일 추가

YeowonKIM commented on src/main/java/com/umc/ttg/domain/coupon/entity/Coupon.java:
```diff
@@ -0,0 +1,38 @@
+    private LocalDate end_date;
+    @Column(nullable = false, length = 1)
+    @ColumnDefault("n")
+    private byte[] status_yn;
```
[2]
byte 배열로 하신 이유가 따로 있을까요?? 
값이 y 또는 n으로 들어가니까 String이나 char이 좋지 않을까 생각해 봤습니다!

5nam replied:
[링크](https://firstws.tistory.com/entry/MySQL-MySQL-%EB%8D%B0%EC%9D%B4%ED%84%B0-%ED%83%80%EC%9E%85%EA%B3%BC-Java-%EB%8D%B0%EC%9D%B4%ED%84%B0-%ED%83%80%EC%9E%85-%EB%B9%84%EA%B5%90%EB%A7%A4%EC%B9%AD)

이 게시글을 보고 char 에 대응하는 것을 써봤는데, char 로 수정하는 것이 더 나을듯하여 수정하겠습니다!

YeowonKIM commented on src/main/java/com/umc/ttg/domain/member/entity/Member.java:
```diff
@@ -0,0 +1,27 @@
+    private String nickname;
+
+    @Column(length = 15)
+    private String phone_num;
```
[3]
자바에서는 변수명, 메서드, 클래스명 등 스네이크 대신 카멜 표기 사용하므로 (+ 코드 컨벤션)
전체적으로 변수명 스네이크 -> 카멜로 변경해야 할 것 같습니다!
(아래 변수명 필요한 부분 표시해 두었습니다)

YeowonKIM commented on src/main/java/com/umc/ttg/domain/review/entity/Review.java:
```diff
@@ -0,0 +1,36 @@
+    private ReviewStatus status;
+
+    @Column(nullable = false)
+    private LocalDate apply_date;
```
[3]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/review/entity/Review.java:
```diff
@@ -0,0 +1,36 @@
+    private Long id;
+
+    @Column(nullable = false)
+    private String review_link;
```
[3]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Region.java:
```diff
@@ -0,0 +1,16 @@
+    @Column(nullable = false, length = 10)
+    private String name;
+
+    private Long upper_id;
```
[3]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Store.java:
```diff
@@ -0,0 +1,45 @@
+
+    private String spon_info;
+
+    private String service_info;
```
use ~ service_info 네 부분 변수병 수정 필요

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Store.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Column(nullable = false, length = 1)
+    @ColumnDefault("n")
+    private byte[] hot_yn;
```
[2]
이 부분도 위에랑 동일 이유로 바이트 배열 대신에
String이나 Char로 하는 게 어떠신가요??

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Store.java:
```diff
@@ -0,0 +1,45 @@
+    private byte[] hot_yn;
+
+    @Column(nullable = false)
+    private int review_count;
```
[3]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/coupon/entity/Coupon.java:
```diff
@@ -0,0 +1,38 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id", nullable = false)
+    private Long id;
```
가독성을 위해 칼럼마다 한 줄씩 띄어쓰는 게 어떠신가요??
(+ 다른 클래스는 모두 칼럼별로 한 줄씩 띄어쓰기가 있어서 통일하면 좋을 것 같습니다)

YeowonKIM commented on src/main/java/com/umc/ttg/domain/review/entity/ReviewStatus.java:
```diff
@@ -0,0 +1,16 @@
+    SUBSCRIPTION("SUBSCRIPTION"),
+    SCREENING("SCREENING"),
+    SUCCESS("SUCCESS"),
+    FAIL("FAIL");
```
이 부분 제가 코드 발급할 때 사용할 것 같은데 
status값 말고 value 하나 더 줘서 상태 순서로 0~3 하나 더 줘서 사용해도 괜찮을까요?
(괜찮다면 제가 코드 작성할 때 수정하겠습니다)

5nam replied:
네, 편하게 수정하셔도 됩니다!

YeowonKIM commented on src/main/java/com/umc/ttg/domain/coupon/entity/Coupon.java:
```diff
@@ -0,0 +1,38 @@
+@Entity
+public class Coupon extends Time {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
[1]
GenerationType을 AUTO보나는 IDENTITY로 두는 게 좋을 것 같다고 생각하는데 어떠신가요?
다른 엔티티들에서도 동일 사유로 GenerationValue 부분을 INDENTITY로 수정하면 좋을 것 같습니다.

• 이유 : AUTO는 해당 필드의 값을 얻기 위해, 이전에 사용했던 데이터를 구하기 때문에 추가적인 DB Connection이 발생.  따라서 Mysql에서는 IDENTITY를 사용하여 insert 시에는 빈 값으로 저장하고 DB에서 값을 지정하는 게 성능 면에서도 좋다고 생각됩니다.

5nam replied:
디테일한 리뷰 감사합니다👍

YeowonKIM commented on src/main/java/com/umc/ttg/domain/member/entity/Member.java:
```diff
@@ -0,0 +1,27 @@
+    private String email;
+
+    @Column(nullable = false)
+    private int benefit_count;
```
[3]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/member/entity/Member.java:
```diff
@@ -0,0 +1,27 @@
+@Entity
+public class Member extends Time {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
[1]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/review/entity/Review.java:
```diff
@@ -0,0 +1,36 @@
+@Entity
+public class Review extends Time {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
[1]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Menu.java:
```diff
@@ -0,0 +1,14 @@
+@Entity
+public class Menu {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
[1]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Region.java:
```diff
@@ -0,0 +1,16 @@
+@Entity
+public class Region {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
[1]

YeowonKIM commented on src/main/java/com/umc/ttg/domain/store/entity/Store.java:
```diff
@@ -0,0 +1,45 @@
+@Entity
+public class Store extends Time {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
[1]

YeowonKIM commented:
전반적으로 연관관계 매핑 잘 된 것 같습니다 고생 많으셨습니다 !
몇 군데 수정하면 좋을 것 같은 부분 작성했는데 그 부분 확인해보시고 괜찮으시면 수정한 뒤 merge하면 좋을 것 같습니다 ☺️

5nam committed:
🎨 개선 Entity
1. byte[] -> char
2. 카멜 표기로 변경
3. 가독성을 위한 칼럼 사이 공백 추가
4. id 에 GenerationType AUTO -> IDENTITY

5nam committed:
🗃 추가 Store 에 빠진 칼럼

1. 상점 테이블에 '서브 타이틀' 추가
2. 상점 테이블에 '리뷰 유지 기간' 추가
3. 상점 테이블에 '가게 장소' 추가
-> 지도를 위한 주소와는 다른 종류. '수유역 8번 출구에서 250m'와 같은 사장님이 따로 작성하는 부분

g00hyun commented:
여원님 피드백 이후 추가적으로 수정필요한 부분은 딱히 없다고 생각됩니다! 추후 개발 진행하면서 추가적으로 수정필요하다 생각하면 의견 공유할게요!
